### PR TITLE
Support multiple server instance connections and store by instance ID

### PR DESCRIPTION
**Pull request by HexaField**
## Summary

- [x] store instances of connections as map/object by ID in state store

## References

closes https://github.com/XRFoundation/XREngine/issues/6078

## Checklist
- [ ] CI/CD checks pass `npm run check`
  - [ ] Linter passing via `npm run lint`
  - [ ] Typescript passing via `npm run check-errors`
  - [ ] Unit & Integration tests passing via `npm run test`
  - [ ] Docker build process passing via `npm run build-client`
- [ ] If this PR is still a WIP, convert to a draft 
- [ ] When this PR is ready, mark it as "Ready for review"
- [ ] Changes have been manually QA'd
- [ ] Changes reviewed by at least 2 approved reviewers

## QA Steps

_List any additional steps required to QA the changes of this PR, as well as any supplemental images or videos._

## Reviewers

_Reviewers for this PR_

### CODE REVIEW

File: packages/client-core/src/components/InstanceChat/index.tsx [hunk 1]
```diff
@@ -55,6 +55,10 @@ const InstanceChat = (props: Props): any => {
   const [unreadMessages, setUnreadMessages] = React.useState(false)
   const activeChannelMatch = Object.entries(channels).find(([key, channel]) => channel.channelType === 'instance')
   const instanceConnectionState = useLocationInstanceConnectionState()
+
+  const currentInstanceId = instanceConnectionState.currentInstanceId.value
+  const currentInstanceConnection = instanceConnectionState.instances[currentInstanceId!]
```
Comment (speigg) [inline packages/client-core/src/components/InstanceChat/index.tsx:60]: is it possible for `instances` to not contain the given `currentInstanceId` ?

Comment (speigg) [inline packages/client-core/src/components/InstanceChat/index.tsx:60]: is it possible for `instances` to not contain the given `currentInstanceId` ?

Comment (HexaField) [reply packages/client-core/src/components/InstanceChat/index.tsx:60]: yes

Comment (speigg) [reply packages/client-core/src/components/InstanceChat/index.tsx:60]: For nullable states, it's generally a good idea to do this:

const currentInstanceConnection = instanceConnectionState.instances[currentInstanceId!].ornull

**Review by barankyle — APPROVED** [review]

**Review by speigg — APPROVED** [review]

**Pull request merged**